### PR TITLE
Fix the deprecation warnings

### PR DESCRIPTION
**Pull request by gmponos**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | yes
| License         | MIT

#### What's in this PR?

@Nyholm I started this PR from your branch. I hope you haven't already been working on it..

The solution is not a very much sophisticated one.. but I gues for now it can do..

### CODE REVIEW

File: src/HttpClientTest.php [hunk 1]
```diff
@@ -103,6 +102,7 @@ public function testSendWithInvalidUri()
             $this->defaultHeaders
         );
 
+        $this->expectException(\Psr\Http\Client\NetworkExceptionInterface::class);
```
Comment (dbu) [inline src/HttpClientTest.php]: please `use` the classes used for these

**Comment by gmponos** [comment]
@dbu checking the initial PR of @Nyholm I think the BC trait is an overkill..  Since Tobias is adding the return types a new major version is needed.. So we can easily drop support for PhpUnit 7 and move to 8 hence the trait is useless.

The alternative would be to apply the sf BC trait here as well: https://github.com/symfony/symfony/blob/v4.4.1/src/Symfony/Bundle/FrameworkBundle/Test/ForwardCompatTestTrait.php


> that is not a change of your PR, but it would be good to see the changes build on a few more versions

I think most of them are failing due to this:
https://github.com/php-http/client-integration-tests/compare/v2.0.0...master#diff-94fd4ca79483c08112a4c7d0e167582dR30

My suggestion would be to finish here.. tag a new version and then check the clients.

**Comment by dbu** [comment]
right, master of client-integration-tests is already designated as version 3 and i think with adding the return type declarations to setUp and tearDown, we should do a new major version of the integration tests.

if we don't support phpunit 7 anymore, clients can't be tested with php 7.1. therefore i think your trait makes sense and we can merge this as is, and clients using the test suite can update to version 3 at their own pace. the forward compatibility trait only avoids the BC break for people using phpunit 7 and overwriting one of our tests setUp method. narrowing the return value to `void` when working with phpunit 7 works fine.

do you want to `use` the expectedException? is there anything else to consider before we merge this into the feature branch?

**Comment by gmponos** [comment]
ok then.. done..

File: src/HttpClientTest.php (hunk 1 above)
Comment (gmponos) [reply src/HttpClientTest.php]: done.

**Pull request merged**